### ISSUE TITLE
version issues.

### ISSUE THREAD

**Issue by yannsay** [impact-initiatives/hypegrammaR#1]
On the latest version, running the 2 following cases gives unexpected results
- using direct_reporting_numerical_categorical gives numbers/summary statistics as NULL and "case not implemented yet"; 
- using group_difference_numerical_categorical gives numbers/summary statistics as NULL, but tests are run; 

Using an older version of hypegrammaR (from someone else's computer) works fine. 

Files to reproduce are here: 
- small script and input files to run the analysis
- 2 versions: latest (current one on Github) and working (one that works).
https://drive.google.com/open?id=12wGTRSPXIe7D7iAlMXWfIOk48z9ucE5_

**Comment by ElliottMess** [comment]
The issue is located within the mean_with_confints_groups function and related functions. The builded version from remotes::install_github shows a condition as follow:

```
  by <- paste0("~", independent.var, sep = "")

  if (!all(is.na(design$variables[[by]]))) {
```
As the `by` variable cannot be found within the design object, it returns a `NULL`.

**However**, this condition does not appear within the source code from the repo.

**Comment by yannsay** [comment]
solved by commit 606875cb01622d870301344e081fb2d01e7b6914
